### PR TITLE
Add option to load overrides for a fileBackedObject

### PR DESCRIPTION
**Pull request by stelar7**

### CODE REVIEW

File: src/DefaultBackedSettings.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+export function defaultBackedObject<T>(path: string, overwritePath: string): T {
+    const defaults = fs.readFileSync(path, "utf-8");
+    const overwrite = fs.readFileSync(overwritePath, "utf-8");
+    const obj = Object.assign({}, JSON.parse(defaults), JSON.parse(overwrite));
+
+    const proxy = {
+        set(object: any, property: string, value: any, receiver: any) {
+            Reflect.set(object, property, value, receiver);
+            try {
+                fs.writeFileSync(path, JSON.stringify(obj));
```
Comment (Querijn) [inline src/DefaultBackedSettings.ts]: Any changes made should probably go to the overwrite path rather than the regular one

File: src/app.ts [hunk 2]
```diff
@@ -15,13 +15,14 @@ import Uptime from "./Uptime";
...
+import { defaultBackedObject } from "./DefaultBackedSettings";
 import { fileBackedObject } from "./FileBackedObject";
 import { PersonalSettings } from "./PersonalSettings";
 import { SharedSettings } from "./SharedSettings";
 
 // Load and initialise settings
-const sharedSettings = fileBackedObject<SharedSettings>("settings/shared_settings.json");
-const personalSettings = fileBackedObject<PersonalSettings>("settings/personal_settings.json");
+const sharedSettings = defaultBackedObject<SharedSettings>("settings/shared_settings.json", "private/shared_settings.json");
+const personalSettings = defaultBackedObject<PersonalSettings>("settings/personal_settings.json", "private/personal_settings.json");
```
Comment (Querijn) [inline src/app.ts]: Merge both setting files together

File: src/DefaultBackedSettings.ts (hunk 1 above)
Comment (stelar7) [reply src/DefaultBackedSettings.ts]: I'm thinking the overwrite path is the changes you always want the same?  
Also, setting a value is most likely never getting used for this, as settings are read-only at the moment

**Comment by Querijn** [comment]
This now has merge conflicts

**Pull request merged**